ui/app/AppLayouts/Profile/views/MyProfileView.qml

```diff
@@ -34,86 +35,221 @@ SettingsContentBase {
+            hasAnyProfileShowcaseChanges = false
+        }
+
+        function save() {
+            if (hasAnyProfileShowcaseChanges)
+                profileStore.storeProfileShowcasePreferences()
+
+            if (!descriptionPanel.isEnsName)
+                profileStore.setDisplayName(descriptionPanel.displayName.text)
+            profileStore.setBio(descriptionPanel.bio.text.trim())
+            profileStore.saveSocialLinks()
+            if (profileHeader.icon === "") {
+                root.profileStore.removeImage()
+            } else {
+                profileStore.uploadImage(profileHeader.icon,
+                                         profileHeader.cropRect.x.toFixed(),
+                                         profileHeader.cropRect.y.toFixed(),
+                                         (profileHeader.cropRect.x + profileHeader.cropRect.width).toFixed(),
+                                         (profileHeader.cropRect.y + profileHeader.cropRect.height).toFixed());
+            }
+
+            reset()
         }
     }
 
     ColumnLayout {
-        id: layout
-        spacing: Constants.settingsSection.itemSpacing
         width: root.contentWidth
 
         StackLayout {
             id: stackLayout
-            currentIndex: editPreviwTabBar.currentIndex
+            currentIndex: profileTabBar.currentIndex
 
-            MyProfileSettingsView {
-                id: settingsView
+            // identity
+            ColumnLayout {
                 objectName: "myProfileSettingsView"
-                profileStore: root.profileStore
-                privacyStore: root.privacyStore
-                walletStore: root.walletStore
-                communitiesModel: root.communitiesModel
-                walletAssetsStore: root.walletAssetsStore
-                currencyStore: root.currencyStore
-
-                onVisibleChanged: if (visible) stackLayout.Layout.preferredHeight = settingsView.implicitHeight
-                Component.onCompleted: stackLayout.Layout.preferredHeight = Qt.binding(() => settingsView.implicitHeight)
+                ProfileHeader {
+                    id: profileHeader
+                    Layout.fillWidth: true
+                    Layout.leftMargin: Style.current.padding
+                    Layout.rightMargin: Style.current.padding
+
+                    store: root.profileStore
+
+                    displayName: profileStore.name
+                    pubkey: profileStore.pubkey
+                    icon: profileStore.profileLargeImage
+                    imageSize: ProfileHeader.ImageSize.Big
+
+                    displayNameVisible: false
+                    pubkeyVisible: false
+                    emojiHashVisible: false
+                    editImageButtonVisible: true
+                }
+
+                ProfileDescriptionPanel {
+                    id: descriptionPanel
+
+                    readonly property bool isEnsName: profileStore.preferredName
```

Comment:
Don't you find it confusing to have the ensName in `preferredName`? Had to go really deep in nim to confirm this is actually the ens

Reply:
Yeah :shrug: